lib/semantics/symbol.cc

```diff
@@ -708,10 +708,22 @@ bool GenericKind::IsOperator() const {
 
 std::string GenericKind::ToString() const {
   return std::visit(
-      common::visitors{
-          [](const OtherKind &x) { return EnumToString(x); },
-          [](const DefinedIo &x) { return EnumToString(x); },
-          [](const auto &x) { return common::EnumToString(x); },
+      common::visitors {
+        [](const OtherKind &x) { return EnumToString(x); },
+            [](const DefinedIo &x) { return EnumToString(x); },
+#if !__clang__ && __GNUC__ == 7 && __GNUC_MINOR__ == 2
+            [](const common::NumericOperator &x) {
+              return common::EnumToString(x);
+            },
+            [](const common::LogicalOperator &x) {
+              return common::EnumToString(x);
+            },
+            [](const common::RelationalOperator &x) {
+              return common::EnumToString(x);
+            },
+#else
+            [](const auto &x) { return common::EnumToString(x); },
+#endif
```

Comment:
It would probably work to use a single lambda with `std::is_same_v` tests. In that case it would be good to comment why `common::visitors` is being avoided.

For example,
```
  return std::visit(
      [](const auto &x) {
        using T = std::decay_t<decltype(x)>;
        if constexpr (std::is_same_v<T, OtherKind> || std::is_same_v<T, DefinedIo>) {
          return EnumToString(x);
        } else {
          return common::EnumToString(x);
        }
      },
      u);
```

Reply:
Yes, I tried it that way as well, and it works, but eventually we'll deprecate GCC 7.2, and after the conditional patch is removed the code will look the way we really want it to look.